### PR TITLE
Find posts by more than one keyword

### LINKED ISSUE #141: Find posts by more than one keyword
Currently, even if a user sends in multiple keywords to find a post, only the first keyword will be used to search for a post. Be able to query for more than one keyword. 

Be sure to search the keywords in both the post title and body.

A/C
- [ ] Determine the number of keywords allowed to search. Discuss with team before making decision. 
- [ ] Determine how to use the keywords in combination. For example, should a post including any keyword be found? Do all keywords need to be in the body or title? Should a more sophisticated algorithm or library be used? Discuss with team before making decision.
- [ ] Enable search by multiple keywords.
- [ ] Remove the TODO comment at https://github.com/OpenEugene/openboard/blob/73e3bf4cb16642bc754c08f5ef5d400dbfdf8213/back/internal/postsvc/internal/postdb/qrypost.go#L111
- [ ] Create tests to ensure the find post query is working as intended.

### PR DESCRIPTION
**Pull request by xipxape**
Closes #141

### CODE REVIEW

**Comment by appins** [comment]
I still need to write tests and we should probably talk about how keywords are actually being used.

File: back/internal/postsvc/internal/postdb/qrypost.go [hunk 1]
```diff
@@ -108,23 +109,20 @@ func (s *PostDB) upsertPost(ctx cx, sid string, x *pb.AddPostReq, y *pb.PostResp
...
 	if err != nil {
 		return err
 	}
 	defer selStmt.Close()
 
-	var like string
-	if len(x.Keywords) > 0 {
-		like = x.Keywords[0]
-	}
+	like := strings.Join(x.Keywords, " ")
```
Comment (appins) [inline back/internal/postsvc/internal/postdb/qrypost.go:120]: Because I don't think that it was intended that they would just be joined together.

**Comment by codegold79** [comment]
I haven't read through the changes carefully, but I wanted to write something here so we don't miss talking about it. Should we limit the number of keywords a person sends to the search? They could purposely overload our database by sending in a novel into the search box, couldn't they?

**Comment by appins** [comment]
> I haven't read through the changes carefully, but I wanted to write something here so we don't miss talking about it. Should we limit the number of keywords a person sends to the search? They could purposely overload our database by sending in a novel into the search box, couldn't they?

@codegold79 Sure. I don't know how big of an issue that could be, but I know Google, for example, only allows searching by the first 30-something words. I'm not sure, however, where we would put that logic. We could put a filter here, but I feel like that belongs somewhere else in the code. I'm not sure, I'm new to this base. 

Also, we may want to limit the number of results a person gets back. I know @coatsd was working on that in another branch, but I'm not sure when/if those changes will be live.

File: back/internal/postsvc/internal/postdb/mysql/0003_add_fulltext_columns.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+-- +migrate Up
```
Comment (daved) [inline back/internal/postsvc/internal/postdb/mysql/0003_add_fulltext_columns.sql:1]: I'd prefer it if the filename was "0003_add_post_fulltext_cols.sql".

**Review by daved — CHANGES_REQUESTED** [review]

File: back/tests/openbsrv/postsvc_test.go [hunk 3]
```diff
@@ -295,6 +296,75 @@ func postSvcFindAllPosts(ctx context.Context, conn *grpc.ClientConn, clnt pb.Pos
...
+		}
+	}
+}
+
+func postSvcKeywordSearch(ctx context.Context, conn *grpc.ClientConn, clnt pb.PostClient) func(*testing.T) {
+	return func(t *testing.T) {
+		tests := []struct {
+			fndReq *pb.FndPostsReq
+			want   *pb.PostsResp
+		}{
```
Comment (daved) [inline back/tests/openbsrv/postsvc_test.go:318]: I'm not sure if we've been following this practice consistently so far... I suggest adding a field for the name of the test. The name is useful to filter for specific tests (when tests are eventually organized to be run independently), and it also documents the nature/intent of the test. For myself, thinking about the nature/name of a test case often brings to mind other test cases that would make sense to nail down.

**Review by daved — CHANGES_REQUESTED** [review]

**Review by daved — APPROVED** [review]

**Review by codegold79 — APPROVED** [review]
LGTM

**Pull request merged**